profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -145,216 +154,283 @@ export function UntraceableState() {
+  const handleSubmit = (e) => {
+    e.preventDefault();
+  };
 
-  return (<div>
-    <input value={name} name="name" type="text" onChange={setName} />
-    <button type="submit" onClick={handleSubmit}>Submit</button>
-  </div>)
+  return (
+    <form onSubmit={handleSubmit}>
+      <input
+        value={name}
+        name="name"
+        type="text"
+        onChange={(e) => setName(e.target.value)}
+      />
+      <button type="submit">Submit</button>
+    </form>
+  );
 }
 
 export function CrudeStateManagement() {
-  const [name, setName] = useState("")
-  const [age, setAge] = useState("")
-  const [location, setLocation] = useState("")
-  const [email, setEmail] = useState("")
-  const [password, setPassword] = useState("")
-  
-
-  const handleSubmit = (e) => {}
-
-  return (<form onSubmit={handleSubmit}>
-    <input value={name} name="name" type="text" onChange={setName} />
-    <input value={age} name="age" type="number" onChange={setAge} />
-    <input value={location} name="location" type="text" onChange={setLocation} />
-    <input value={email} name="email" type="email" onChange={setEmail} />
-    <input value={password} name="password" type="password" onChange={setPassword} />
-    <button type="submit">Submit</button>
-  </form>)
+  const [formState, setFormState] = useState({
+    name: "",
+    age: "",
+    location: "",
+    email: "",
+    password: "",
+  });
+
+  function handleFormValueChange(e) {
+    setFormState({
+      ...formState,
+      [this.name]: this.value,
+    });
+  }
+
+  const handleSubmit = (e) => {};
+  const { name, age, location, email, password } = formState;
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <input
+        value={name}
+        name="name"
+        type="text"
+        onChange={handleFormValueChange}
+      />
+      <input
+        value={age}
+        name="age"
+        type="number"
+        onChange={handleFormValueChange}
+      />
+      <input
+        value={location}
+        name="location"
+        type="text"
+        onChange={handleFormValueChange}
+      />
+      <input
+        value={email}
+        name="email"
+        type="email"
+        onChange={handleFormValueChange}
+      />
+      <input
+        value={password}
+        name="password"
+        type="password"
+        onChange={handleFormValueChange}
+      />
+      <button type="submit">Submit</button>
+    </form>
+  );
 }
 
 export function UnidiomaticHTMLHierarchy() {
-  const bids = [1,2,3]
-  const asks = [1,2,3]
+  const bids = [1, 2, 3];
+  const asks = [1, 2, 3];
 
-  return (<li>
-    {bids.map((bid, i) => <span key={i}>{bid}</span>)}
-    {asks.map((ask, j) => <span key={j+'asks'}>{ask}</span>)}
-  </li>)
+  return (
+    <li>
+      {bids.map((bid, i) => (
+        <span key={i}>{bid}</span>
+      ))}
+      {asks.map((ask, j) => (
+        <span key={j + "asks"}>{ask}</span>
+      ))}
+    </li>
+  );
 }
 
 export function SubstandardDataStructure() {
```

Comment:
This one I don't know either, are we supposed to have an error stack? error object with message?